app/lib/widgets/app_layout.dart

```diff
@@ -0,0 +1,234 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:threebotlogin/helpers/globals.dart';
+
+/// Main app layout wrapper that handles navigation, drawer, app bar, and bottom navigation
+/// This replaces the layout logic that was previously in HomeScreen
+class AppLayout extends StatefulWidget {
+  const AppLayout({
+    super.key,
+    required this.child,
+    this.showAppBar = true,
+    this.showDrawer = true,
+    this.showBottomNav = true,
+    this.appBarActions,
+    this.title,
+    this.onAddFarm,
+    this.onAddWallet,
+  });
+
+  final Widget child;
+  final bool showAppBar;
+  final bool showDrawer;
+  final bool showBottomNav;
+  final List<Widget>? appBarActions;
+  final String? title;
+  final VoidCallback? onAddFarm;
+  final VoidCallback? onAddWallet;
+
+  @override
+  State<AppLayout> createState() => _AppLayoutState();
+}
+
+class _AppLayoutState extends State<AppLayout> {
```

Comment:
- I suggest creating simple data class to hold icon, label and index for each navigation entry to be used anywhere.

Reply:
Done.